### PR TITLE
feature(carousel): carousel-item interval

### PR DESCRIPTION
**Pull request by morrissey-ingenious**
adds the ability to assign data-interval to an individual carousel-item.

I'm aware this may not be the proper way to add this functionality so please comment and give feedback and suggestions so that I can modify this pull request.

thanks

### CODE REVIEW

File: js/src/carousel.js [hunk 1]
```diff
@@ -385,6 +385,20 @@ const Carousel = (($) => {
         $(activeElement).addClass(directionalClassName)
         $(nextElement).addClass(directionalClassName)
 
+        if ($(nextElement).data('interval')) {
```
Comment (Johann-S) [inline js/src/carousel.js]: Please use more vanilla JS and change that line by
```js
const nextElementInterval = nextElement.getAttribute('data-interval')
if (nextElementInterval) {
```

File: js/src/carousel.js [hunk 2]
```diff
@@ -385,6 +385,20 @@ const Carousel = (($) => {
...
+          Util.typeCheckConfig(NAME,
+            {
+              interval : $(nextElement).data('interval')
+            },
+            {
+              interval : '(number|boolean)'
+            })
+          this._config.defautlInterval = this._config.defautlInterval || this._config.interval
```
Comment (Johann-S) [inline js/src/carousel.js]: typo here, it's not `defautlInterval` but `defaultInterval`

File: js/src/carousel.js [hunk 3]
```diff
@@ -385,6 +385,20 @@ const Carousel = (($) => {
...
+          this._config.interval = $(nextElement).data('interval')
```
Comment (Johann-S) [inline js/src/carousel.js]: Here instead of get the value from the dom use `nextElementInterval`

**Review by Johann-S — CHANGES_REQUESTED** [review]

**Comment by Johann-S** [comment]
There are a lots of things to do before we accept this MR :
- please add a CodePen which demonstrate what your code is doing
- add a unit test
- update our documentation

**Comment by morrissey-ingenious** [comment]
@Johann-S what is the best way to pull these changes into a CodePen to show what my code is doing?  Thank You

**Review by Johann-S — APPROVED** [review]

**Comment by Johann-S** [comment]
very nice @morrissey-ingenious 👍 

I want the feedbacks from other members before merging your PR

/CC @XhmikosR, @patrickhlauke, @mdo

**Review by XhmikosR — APPROVED** [review]

**Pull request merged**